### ISSUE TITLE
NoClassDefFoundError after editing the Person class property (AB3)

### ISSUE THREAD

**Issue by darryl-chan** [nus-cs2103-AY2324S2/forum#466]
I changed the person class property and removed address and telephone number and added student id and telegram handle as their property. This caused me to change the class names in the main and test units.

After fixing the code I managed to get the main running, which means that there isnt any compilation error for main.

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/113418549/5161dd1a-c441-4e6e-b8d3-06669ed97f1d)

However, when I build with Gradle, the test units are failing with a message that I do not understand.

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/113418549/a1992adb-1fa5-4462-95d0-473188ecd2b7)

I have changed all the class names and ensured that they are correct in the test units, i.e. there wasn't any compilation error.

I've checked online and it seems that the issue is with the class path but the two properties I added are in the same file as person where address and phone was before. 

Has anyone faced this issue? Any help appreciated.

**Comment by damithc** [comment]
@darryl-chan It might help if you create a draft PR in your team repo and share the link here, so that others can see what exactly you did in the code.

**Comment by darryl-chan** [comment]
[PR link](https://github.com/darryl-chan/tp/pull/2)

**Comment by delishad21** [comment]
Hello, seems like I found the issue.

Under `TypicalPersons.java` in `seedu.address.testutil` on lines 48 and 49,

![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/103935416/34b73d41-842e-4d05-9d52-ec56a370916d)

You initialised `IDA` with an invalid telegram handle `idaho`. Your validation code requires your telegram handles to start with `@` so this leads to an `IllegalArgumentException`.

I'm not extremely clear on this but I believe what ends up happening is that the TypicalPerson class does not get initialised properly, and any subsequent references to the static values of the class below 'IDA' results in the `NoClassDefFoundError` you see in your tests

Changing `idaho` to `@idaho` seems to stop all `NoClassDefFoundError` related testcases from failing. 

On a side note, I was able to trace down the source of the error by running the gradle test with `--info` which gave me this additional information.
![image](https://github.com/nus-cs2103-AY2324S2/forum/assets/103935416/de98a7dc-b1d8-406b-a18d-953c260a82a4)
